File: laravel-12-ddd-standard/examples/simple-service/src/Application/UseCases/RegisterObraUseCase.php
<?php
declare(strict_types=1);

namespace App\Examples\SimpleService\Application\UseCases;

use App\Examples\SimpleService\Domain\Entities\Obra;
use App\Examples\SimpleService\Domain\Repositories\ObraRepositoryInterface;

final class RegisterObraUseCase
{
    private ObraRepositoryInterface $repository;

    public function __construct(ObraRepositoryInterface $repository)
    {
        $this->repository = $repository;
    }

    public function execute(array $data): Obra
    {
        $id = $this->generateId();
        $titulo = (string) ($data['titulo'] ?? '');
        $descripcion = isset($data['descripcion']) ? (string) $data['descripcion'] : null;

        $obra = new Obra($id, $titulo, $descripcion);

        $this->repository->save($obra);

        return $obra;
    }

    private function generateId(): string
    {
        return bin2hex(random_bytes(8));
    }
}

File: laravel-12-ddd-standard/dotnet-skeleton/src/Application/UseCases/RegisterObraUseCase.cs
using System;
using System.Threading.Tasks;

namespace App.DotNet.Skeleton.Application.UseCases
{
    using App.DotNet.Skeleton.Domain.Entities;
    using App.DotNet.Skeleton.Domain.Repositories;

    public sealed class RegisterObraUseCase
    {
        private readonly IObraRepository _repository;

        public RegisterObraUseCase(IObraRepository repository)
        {
            _repository = repository;
        }

        public async Task<Obra> ExecuteAsync(string titulo, string? descripcion = null)
        {
            var id = GenerateId();
            var obra = new Obra(id, titulo, descripcion);
            await _repository.SaveAsync(obra).ConfigureAwait(false);
            return obra;
        }

        private static string GenerateId() => Guid.NewGuid().ToString("N");
    }
}

